File: MedArrival-Front/src/models/ProductDTO.ts
import { BaseDTO } from "./BaseDTO";
import { PriceComponentDTO } from "./PriceComponentDTO";

export interface ProductDTO extends BaseDTO {
  name: string;
  description: string;
  priceComponents: PriceComponentDTO[];
  availableQuantity: number;
  totalCost: number;
}

File: MedArrival-Backend/src/main/java/com/cnesten/medarrivalbackend/DTO/ProductDTO.java
package com.cnesten.medarrivalbackend.DTO;

import lombok.Data;

import java.time.LocalDateTime;
import java.util.List;

@Data
public class ProductDTO {
    private Long id;
    private String name;
    private String description;
    private Integer availableQuantity;
    private ProductCategoryDTO category;
    private List<PriceComponentDTO> priceComponents;
    private Float totalCost;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private String createdBy;
    private String updatedBy;
}
